Spuštění pouze některých podúloh

filip-hejsek opened the pull request:
Tento PR přidá parametr `--subtasks` příkazu `run`:
```bash
$ ./ksp-klient.py run 33-3-3 --subtasks 3 ./my_program # Spustí my_program na podúlohu 3
$ ./ksp-klient.py run 33-3-3 --subtasks 1,2 ./my_program # Spustí my_program na podúlohu 1 a 2
$ ./ksp-klient.py run 33-3-3 ./my_program # Spustí my_program na všechny podúlohy (stejně jako předtím)
```
Bez tohoto je možné pouze buď spustit všechny podúlohy, nebo manuálně stáhnout vstup, spustit program a nahrát výstup.

Zbývající otázky:
 - Možná bych ještě přidal zkratku, může to být třeba `-s`?

filip-hejsek committed:
Přidán parametr --subtasks příkazu run

gollux commented:
To se mi líbí, `-s` zní dobře -- doděláš, prosím?

filip-hejsek committed:
Přidána zkratka -s parametru --subtasks

trueMiskin commented:
O tom jsem již přemýšlel, ale když už jsme u vyjmenovávání, tak proč nepoužít intervaly, např. `1-4,6-7 ` mě pouhé vyjmenovávání přijde zbytečně omezující. Co na to @gollux ?

filip-hejsek commented:
> O tom jsem již přemýšlel, ale když už jsme u vyjmenovávání, tak proč nepoužít intervaly, např. `1-4,6-7 ` mě pouhé vyjmenovávání přijde zbytečně omezující.

Také mě to napadlo, ale říkal jsem si, že je to příliš složité a vyžadovalo by to delší vysvětlení v nápovědě. Ale myslím, že by to bylo užitečné.

gollux commented:
V nápovědě by myslím stačil příklad. Souhlasím, že by to bylo hezké :)

filip-hejsek committed:
Přidána podpora intervalů v parametru --subtasks

Velikost intervalů je omezená na 1000000, protože se celý seznam ukládá
do paměti.

Zároveň byly zakázány duplicitní podúlohy.

filip-hejsek commented:
Přidal jsem podporu intervalů

trueMiskin commented on ksp-klient.py:
```diff
@@ -330,7 +330,25 @@ def example_usage(text: str) -> str:
+        else:
+            result.append(int(range_string))
+    if len(result) > len(set(result)):
+        raise argparse.ArgumentTypeError("seznam obsahuje duplicitní hodnoty")
```
Myslím, že by se s tímto případem měl program vypořádat. Buď udělá přesně, co uživatel chtěl, tedy vyřeší nějaký problém víckrát. Nebo vstup hodí do setu a vstupy třeba poté setřídí, aby vstupy nebyly seřazené náhodně. (nebo klidně i nějaké jiné pracnější chování)

První může být použitelné, pokud chceme zachovat pořadí třeba kvůli přepínači, který při první špatné odpovědi se klient ukončí.
Druhé zase eliminuje opakované dotazování.

gollux replied:
Já myslím, že duplicitní hodnoty ničemu nevadí. Jen bych vynechal kontrolu.

filip-hejsek replied:
Odebral jsem kontrolu duplicitních hodnot.

filip-hejsek committed:
Odebrána kontrola duplicitních hodnot
